### PR TITLE
set fake strain options, sample rate, filter settings, etc

### PR DESCRIPTION
**Pull request by ahnitz**

### CODE REVIEW

**Comment by ahnitz** [comment]
This adds 3 options to the strain generation 

```
 --fake-strain-flow FAKE_STRAIN_FLOW
                        Low frequency cutoff of the fake strain
  --fake-strain-filter-duration FAKE_STRAIN_FILTER_DURATION
                        Duration in seconds of the fake data coloring filter
  --fake-strain-sample-rate FAKE_STRAIN_SAMPLE_RATE
                        Sample rate of the fake data generation
```

There are defaults for these similar to the current ones.

**Comment by ahnitz** [comment]
@titodalcanton @spxiwh This also removes the strangely hidden dependence on an 'low-frequency-cutoff' option and makes it explicit for generating fake data. It also has a default which will be sufficient for current LIGO data.

**Comment by ahnitz** [comment]
Note: All fake data will now be different after this patch.

File: pycbc/noise/reproduceable.py [hunk 1]
```diff
@@ -26,39 +26,41 @@
 from pycbc.types import TimeSeries, FrequencySeries, complex_same_precision_as
 from numpy.random import RandomState
 
-# These need to be constant to be able to recover identical results.
-# The hope is that nobody needs a higher resolution
```
Comment (spxiwh) [inline pycbc/noise/reproduceable.py:30]: Well, that hope didn't last ;-)

**Review by spxiwh — CHANGES_REQUESTED** [review]
So I'm going to request an addition to the test suite alongside this. Can we add a test where we generate a small number of Gaussian noise sets (testing the various options that this module has) and check it's the same as a set of values stored in a HDF file (also added with this patch). 

I also note that if you had the block size as the old value of 1638400 you would produce the old output, although it's a weird number!

**Comment by ahnitz** [comment]
> So I'm going to request an addition to the test suite alongside this. Can we add a test where we generate a small number of Gaussian noise sets (testing the various options that this module has) and check it's the same as a set of values stored in a HDF file (also added with this patch).
> 
> I also note that if you had the block size as the old value of 1638400 you would produce the old output, although it's a weird number!

This is a fair request. I'll think about it. I'm not sure I want us to be committing any more 'files' in the repo, but one might be able to do a checksum or similar for this kind of test. In addition, I think checking that the psd of the data matches the input psd might be a sensible sanity check.

**Comment by spxiwh** [comment]
> > So I'm going to request an addition to the test suite alongside this. Can we add a test where we generate a small number of Gaussian noise sets (testing the various options that this module has) and check it's the same as a set of values stored in a HDF file (also added with this patch).
> > I also note that if you had the block size as the old value of 1638400 you would produce the old output, although it's a weird number!
> 
> This is a fair request. I'll think about it. I'm not sure I want us to be committing any more 'files' in the repo, but one might be able to do a checksum or similar for this kind of test. In addition, I think checking that the psd of the data matches the input psd might be a sensible sanity check.

I like the checksum idea, but we need to be careful that we expect byte-equal output. If for e.g. any FFT is involved in producing the PSD or the data, we'd need to ensure it is reproducible (use FFTW always with measurelvl=0 *should* do it)

**Comment by ahnitz** [comment]
@spxiwh Ok, please look at this again. I've added a simple unit test at the moment that does mostly what we discussed. It's not really a checksum but likely close enough for what we want.

**Review by spxiwh — APPROVED** [review]

**Pull request merged**